lib/slacks/connection.rb

```diff
@@ -273,23 +272,27 @@ def get_dm_for_username(name)
 
     def get_dm_for_user_id(user_id)
       user_ids_dm_ids[user_id] ||= begin
-        response = api("im.open", user: user_id)
+        response = api("conversations.open", user: user_id)
         response["channel"]["id"]
       end
     end
 
 
 
     def fetch_channels!
-      response = api("channels.list")
-      @channels_by_id = response["channels"].index_by { |attrs| attrs["id"] }
-      @channel_id_by_name = Hash[response["channels"].map { |attrs| ["##{attrs["name"]}", attrs["id"]] }]
+      response = api("conversations.list")
+      channels, rest = response["channels"].partition { |attrs| attrs["is_channel"] }
+      groups, ims = rest.partition { |attrs| attrs["is_group"] }
+      @groups_by_id = groups.index_by { |attrs| attrs["id"] }
+      @group_id_by_name = Hash[groups.map { |attrs| [attrs["name"], attrs["id"]] }]
+      user_ids_dm_ids.merge! Hash[ims.map { |attrs| attrs.values_at("user", "id") }]
+      @channels_by_id = channels.index_by { |attrs| attrs["id"] }
+      @channel_id_by_name = Hash[channels.map { |attrs| ["##{attrs["name"]}", attrs["id"]] }]
     end
 
     def fetch_groups!
-      response = api("groups.list")
-      @groups_by_id = response["groups"].index_by { |attrs| attrs["id"] }
-      @group_id_by_name = Hash[response["groups"].map { |attrs| [attrs["name"], attrs["id"]] }]
+      fetch_channels!
+      @group_id_by_name
```

Comment:
Why keep `fetch_groups!` around at all anymore? Would it make more sense to rename `fetch_channels!` to `fetch_conversations!` now?

Reply:
Agreed on both counts. I'll make those changes. 👍